src/nvim/lua/vim.lua

```diff
@@ -431,7 +431,9 @@ function vim.region(bufnr, pos1, pos2, regtype, inclusive)
   local bufline
   if regtype:byte() == 22 then
     bufline = vim.api.nvim_buf_get_lines(bufnr, pos1[1], pos1[1] + 1, true)[1]
-    pos1[2] = vim.str_utfindex(bufline, pos1[2])
+    if pos1[2] < #bufline then
+      pos1[2] = vim.str_utfindex(bufline, pos1[2])
```

Comment:
hmm this can't be right to start with, must use `virtcol()` or something (which should handle virtualedit by itself)

Reply:
That shouldn't be there, a commit from a different PR must have gotten in when rebasing and force-pushing. I'll drop it.

Thanks!